android/java/org/chromium/chrome/browser/crypto_wallet/observers/TxServiceObserverImpl.java

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.crypto_wallet.observers;
+
+import org.chromium.brave_wallet.mojom.TransactionInfo;
+import org.chromium.mojo.system.MojoException;
+
+public class TxServiceObserverImpl implements org.chromium.brave_wallet.mojom.TxServiceObserver {
+    public interface TxServiceObserverImplDelegate {
+        default void onNewUnapprovedTx(TransactionInfo txInfo) {}
+        default void onUnapprovedTxUpdated(TransactionInfo txInfo) {}
+        default void onTransactionStatusChanged(TransactionInfo txInfo) {}
+    }
+
+    private TxServiceObserverImplDelegate mDelegate;
+
+    public TxServiceObserverImpl(TxServiceObserverImplDelegate delegate) {
+        mDelegate = delegate;
+    }
+
+    @Override
+    public void onNewUnapprovedTx(TransactionInfo txInfo) {
+        if (mDelegate == null) return;
```

Comment:
nit: can replace `if (mDelegate == null) return;` checks with a method `isDisconnected`

Reply:
I don't think it's efficient to move one string to a separate function.

Reply:
one string is that `if(isDisconnected())return;`. What is the point to create a function for one string?

Reply:
I don't see a point to blow out a class with 1 line functions. That makes sense if you have a complex condition check or there is something extra has to be done.